xarray/conventions.py

```diff
@@ -279,8 +271,6 @@ def encode_cf_datetime(dates, units=None, calendar=None):
 
     if units is None:
         units = infer_datetime_units(dates)
-    else:
```

Comment:
I agree with you that we shouldn't change units that are user specified, and I'm no longer sure what I was thinking here. Unfortunately the original commit (https://github.com/pydata/xarray/commit/14466e68f1ccdbaf428833aec04d8067379365a4) doesn't really help either.

If our test suite passes, I think removing should be fine.